wheatley/parsing.py

```diff
@@ -6,11 +6,53 @@
 import logging
 
 from wheatley.aliases import CallDef, JSON
+from wheatley.bell import Bell
 from wheatley.row_generation import RowGenerator
 from wheatley.row_generation.place_notation_generator import PlaceNotationGenerator
 from wheatley.row_generation.complib_composition_generator import ComplibCompositionGenerator, \
                                                                   PrivateCompError, InvalidCompError
 
+class StartRowParseError(ValueError):
+    """
+    An error thrown with a helpful error message when the user inputs a start row string that
+    cannot be parsed by the parser.
+    """
+
+    def __init__(self, start_row_string: str, message: str) -> None:
+        super().__init__()
+
+        self.start_row_string = start_row_string
+        self.message = message
+
+    def __str__(self) -> str:
+        return f"Error parsing start row '{self.start_row_string}': {self.message}"
+
+def parse_start_row(start_row: str) -> int:
```

Comment:
I think this should return an entirely parsed `Row` - the currently returned value will be `len(row)`.  Also, the contents of `generate_starting_row` should be combined with this function because if we return a `Row` then there's nothing to generate.

Reply:
We couldn't use a row returned from this directly, as it could be extended by either row gen or bot if they need more bells.
That in turn has a risk of manipulating the same list object.

Reply:
Yeah, I agree with @centreboard, this covers a few of your comments.

The issue with fully parsing and passing in a row is that we don't actually know how many bells there are.
We need to re-generate when we actually start ringing (or tower size changes) and to do that you need the original string (or some mutable object but that seems worse)

Because of that I didn't see the point in generating the row just to add an extra property that would not be used in the end.